src/http/http_parser.h

```diff
@@ -432,3 +434,128 @@ namespace http
     }
   };
 }
+
+namespace ws
```

Comment:
Can these move to their own files?

Reply:
Sure, there's some slight awkwardness with HTTP* things being either HTTP as opposed to WS or HTTP the superset of HTTP and WS, including the upgrade path. I do not have a clever naming idea to resolve this though.